zigup.zig

```diff
@@ -508,37 +478,53 @@ fn cleanCompilers(allocator: *Allocator) !void {
     defer install_dir.close();
     const master_points_to_opt = try getMasterDir(allocator, &install_dir);
     defer if (master_points_to_opt) |master_points_to| allocator.free(master_points_to);
-    var it = install_dir.iterate();
-    while (try it.next()) |entry| {
-        if (entry.kind != .Directory)
-            continue;
+    if (compiler_name_opt) |compiler_name| {
         if (default_comp_opt) |default_comp| {
-            if (mem.eql(u8, default_comp, entry.name)) {
-                std.debug.warn("keeping '{}' (is default compiler)\n", .{default_comp});
-                continue;
+            if (mem.eql(u8, default_comp, name)) {
```

Comment:
We should be able to share the code that checks for default compiler and master.